service/geth/geth.go

```diff
@@ -299,28 +291,16 @@ func (gopts GethOpts) StaticNodesFile() string {
 	return filepath.Join(gopts.Datadir, "/static-nodes.json")
 }
 
-func chainParamsFromGenesisFile(genesisPath string) *chain.ChainParameters {
+func chainConfigFromGenesisFile(genesisPath string) *params.ChainConfig {
 	data, err := ioutil.ReadFile(genesisPath)
 	if err != nil {
 		log.Crit("Can't read genesis.json on DataDir", "genesisPath", genesisPath, "err", err)
 	}
 
-	// We only map the fields we need
-	var genesis struct {
-		Config struct {
-			ChainId  uint64 `json:"chainId"`
-			Isntabul struct {
-				Epoch uint64 `json:"epoch"`
-			} `json:"istanbul"`
-		} `json:"config"`
-	}
+	var genesis core.Genesis
 
 	if err = json.Unmarshal(data, &genesis); err != nil {
 		log.Crit("Can't parse genesis.json on DataDir", "genesisPath", genesisPath, "err", err)
 	}
-
-	return &chain.ChainParameters{
-		ChainId:   new(big.Int).SetUint64(genesis.Config.ChainId),
-		EpochSize: genesis.Config.Isntabul.Epoch,
-	}
+	return genesis.Config
```

Comment:
Can only comment here, but should we bail out if the above unmarshalling fails?

Reply:
I believe the `log.Crit` above should cover this, or is there some situation where unmarshal would fail for our purposes but not return an error?

Reply:
Ah, wasn't aware that `Crit` quits.